### DISCUSSION TITLE
Restrict access to group management in administration console

### DISCUSSION THREAD

**Discussion by pepesoriagarcia99** [keycloak/keycloak#16769] [Archive]
Hi,
We have a multi-tenant solution with one master realm and a realm for each customer.

We have a User Admins who should create and manage users for realms assigned to them, i.e. Only have access to the User-section. We don’t want User Admins to access the Group-section as we preset all groups, so we don’t want anyone to mess with those.

Our User Admins don’t have roles for the master realm’s admin console.

We create a group for each customer realm with roles View User and Manage User that we assign to our User Admins.

This way our User Admins can sign in to the admin console and only browse the customers they are assigned.

When they select a customer they see a restricted admin console showing only Groups and Users section.

But this is not what we want, we want ONLY Users section to be available, so how do hide the Groups section.

Thanks in advance.

**Comment by jonkoops** [comment]
@ssilvert I think we might be hiding some parts of the UI where folks do have access to view certain data, but not to edit it. In such cases I believe the old console would present these forms as read only (disabled fields and no save button).

@pepesoriagarcia99 could you enable the old UI and let us know if what you want to do is possible there? You can do this under 'Realm settings :arrow_right: 'Themes', under 'Admin UI theme' select 'keycloak'.

**Comment by pepesoriagarcia99** [comment]
When we generate a user with the realm roles to view and manage users, it implicitly contains the permissions to manage groups (view - edit - delete).

I attach example images.

roles assigned to the user who must manage user creation:
![image](https://user-images.githubusercontent.com/45427118/216109575-df5ab04b-ed69-43a1-8def-8c7daaeff709.png)

Platform viewed by said user:
![image](https://user-images.githubusercontent.com/45427118/216110090-09a63c75-519d-4b29-92dd-676561c62946.png)

we are using version 20.0.1
thanks in advance @jonkoops

**Comment by ssilvert** [comment]
The old UI does this the same way.  As you said, the `view-users` role has `query-groups` as an associated role.  This tells the UI to display the groups section.

You can remove `query-groups` as an associated role by going to Clients --> realm-management --> Roles --> view-users.

If you do that then the `Groups` won't show up on the left side navigation. You can try that now.  However, the `Groups` tab will still show up for individual users.  To hide that we will need to tweak the code.

Note that your users would still be able to do group assignments if they use the REST API.  We would have to change the back end to lock down the REST API and I don't think we want to change it.  But I don't mind making the needed change to the admin UI that just hides the tab.

Would that solve your problem?

**Comment by pepesoriagarcia99** [comment]
Ok, I understand.

Thank you very much @ssilvert and @jonkoops

**Comment by ssilvert** [comment]
@pepesoriagarcia99 Would my solution solve your use case?

**Reply by pepesoriagarcia99** [reply]
yes, thanks

**Comment by ssilvert** [comment]
OK.  We are making the change.

**Reply by BradBach** [reply]
@ssilvert was this change ever completed and pushed to production?

**Reply by ssilvert** [reply]
@off-whyte That was a long time ago.  I assume that we did follow up on it.  If you are having a problem, please open a new issue.
